include/picongpu/traits/GetMargin.hpp

```diff
@@ -31,10 +31,10 @@ namespace picongpu
          * \tparam Solver solver which needs ghost cells for solving a problem
          *         if solver not define `LowerMargin` and `UpperMargin` this trait (GetMargin)
          *         must be specialized
-         * \tparam SubSetName a optional name (id) if solver needs different ghost cells
+         * \tparam T_Parameter an optional parameter type
          * for different objects
          */
-        template<class Solver, unsigned int SubSetName = 0>
+        template<class Solver, typename T_Parameter = void>
```

Comment:
I will modernize this file and add `T_Parameter` to `GetLowerMargin` and `GetUpperMargin` as a follow-up PR.

Reply:
Unrelated note: there were no other use cases for the integer template parameter, and anyhow type parameter is just more general, one could use an integral constrant type if needed there.

Reply:
The modernization submitted in #3564.